cluster-autoscaler/cloudprovider/gce/mig_info_provider.go

```diff
@@ -43,7 +44,7 @@ type MigInfoProvider interface {
 	// GetMigBasename returns basename for given MIG ref
 	GetMigBasename(migRef GceRef) (string, error)
 	// GetMigInstanceTemplateName returns instance template name for given MIG ref
-	GetMigInstanceTemplateName(migRef GceRef) (string, error)
+	GetMigInstanceTemplateName(migRef GceRef) (InstanceTemplateNameType, error)
 	// GetMigInstanceTemplate returns instance template for given MIG ref
 	GetMigInstanceTemplate(migRef GceRef) (*gce.InstanceTemplate, error)
```

Comment:
Please rebase your changes on the current master. We've recently added new logic around caching the kube-env for templates (https://github.com/kubernetes/autoscaler/pull/6531) which also relies on storing the template naming to work properly.

Reply:
Done